### PR TITLE
Feature: Run callbacks on through model when detaching association

### LINKED ISSUE #2512: Detaching a record via an association calls delete_all, skipping ActiveRecord callbacks, instead of destroy.
### Describe the bug
Detaching an association record calls `delete_all` instead of `destroy` or `destroy_all`. This skips any ActiveRecord callbacks that may exist on the association join record.

### Steps to Reproduce
1. Setup some basic rails models like the example below for a top level model called `Facility`.
2. Generate the Avo resources and add the has_many associations so that you can attach/detach `facilities` on the `user` resource.
3. Detach a `facility` from a `user`. The callback `before_destroy` is not hit within `FacilityUser`. Rails console logs also show _FacilityUser Delete All_ instead of something like _FacilityUser Destroy_.

### Expected behavior & Actual behavior
Although `delete_all` is more performant, we would expect `destroy` or `destroy_all` to be called upon detaching so that any ActiveRecord callbacks are hit. Is there a reason that deletion was chosen?

### Models and resource files

```rb
# Facility
has_many :facility_users, dependent: :destroy
has_many :users, through: :facility_users

# User
has_many :facility_users, dependent: :destroy
has_many :facilities, through: :facility_users

# FacilityUser
before_destroy :do_something

belongs_to :facility
belongs_to :user

def do_something
 ...
end
```

### System configuration
**Avo version**: 3.4.1
**Rails version**: 7.0.8
**Ruby version**: 3.2.2

**License type**:
 - [ ] Community
 - [ ] Pro
 - [x] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [ ] High impact (It makes my app un-usable.)
 - [x] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [x] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

### PR DESCRIPTION
**Pull request by binarygit**
# Description

Fixes #2512

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [x] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Comment by binarygit** [comment]
@Paul-Bob 
```
# team.rb
  has_many :team_members, through: :memberships, class_name: "User", source: :user, inverse_of: :teams

# user.rb
  has_and_belongs_to_many :teams, join_table: :team_memberships, inverse_of: :admin
```

Here team defines it's team_members assoc using a through table however, User defines the same association as a HABTM.

Detaching a user from the team show page will call `team_membership.destroy` but because user has setup this assoc as HABTM I don't think when detaching a team from the user's show page should call `team_membership.destroy`. Am I correct?

**Comment by Paul-Bob** [comment]
> Update: Adrian and I talked about this. user.rb should be declaring the relationship as a through relationship. It's unusual to declare a relationship as a through at one end and a HABTM on the other end.

Agree, I also noticed that particularity on our dummy recently. 

Does this PR work well from both sides if the associations are correctly defined?

File: spec/features/avo/has_many_field_spec.rb [hunk 1]
```diff
@@ -154,4 +154,29 @@
...
+      team.team_members << user
+      expect(team.team_members.count).to eq 1
+      team_membership = team.memberships.first
+
+      visit "/admin/resources/teams/#{team.id}/team_members?view=show&turbo_frame=has_many_field_show_team_members"
+
+      expect { find("tr[data-resource-id='#{user.to_param}'] [data-control='detach']").click }.to raise_error("Callback Called")
+    end
+
+    it "triggers callbacks when called from the other model too" do
```
Comment (binarygit) [inline spec/features/avo/has_many_field_spec.rb:172]: Yes it works both ways of the association @Paul-Bob. This test verifies that :sweat_smile:

File: spec/dummy/app/models/user.rb [hunk 2]
```diff
@@ -38,7 +38,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :team_memberships
   has_and_belongs_to_many :projects, inverse_of: :users
-  has_and_belongs_to_many :teams, join_table: :team_memberships, inverse_of: :admin
+  has_many :team_memberships
```
Comment (Paul-Bob) [inline spec/dummy/app/models/user.rb:41]:
```suggestion
```

Same association already defined (L39)

**Review by Paul-Bob — CHANGES_REQUESTED** [review]
Looking good! Thanks @binarygit, let's please remove the duplicated association and also solve the conflicts.

**Comment by binarygit** [comment]
Wait. I’m working on displaying the messages when associations are misconfigured

File: spec/dummy/app/models/user.rb (hunk 2 above)
Comment (binarygit) [reply spec/dummy/app/models/user.rb:40]: Thanks. I didn't notice this!

**Comment by binarygit** [comment]
@Paul-Bob I couldn't find a straightforward way to check if associations are properly configured. Sorry for holding this up, maybe merge this now and I'll pick this up again later.

**Comment by binarygit** [comment]
I actually talked about this here: https://rubyonrails-link.slack.com/archives/C05054QPL/p1721928807967489 and I don't think verifying associations are setup correctly should be our job. 

I did try and it's too much work and also feels like it should be upto the user (like the devs point out in the link) to set it up correctly.

**Comment by Paul-Bob** [comment]
It's ok without the association check @binarygit thanks for looking into it

**Review by adrianthedev — APPROVED** [review]
Looks good.

**Review by Paul-Bob — APPROVED** [review]
It's looking good, thanks @binarygit!

**Pull request merged**